docs/ephemeral-testing-w-notifications.md

```diff
@@ -0,0 +1,206 @@
+
+```shell
+# testing Quay
+docker/podman login quay.io -u username@redhat.com
+Password:
+Login Succeeded!
+
+# testing Registry.Redhat.io
+docker/podman login registry.redhat.io -u username@redhat.com
+Password:
+Login Succeeded!
+```
+
+> NOTE: The process for Mac and Linux is slightly different to account for those using ARM laptops. The `build-push-minimal` make target used for Mac forces the image be built for linux/amd64 arch to ensure it can run in the ephemeral cluster. If you are running on an ARM laptop, it is not suitable for running locally
+
+### Building Relations API
+1) Change to the Relations API code path: `cd /path/to/relations-api`
+
+**On Linux**:
+
+2) Set required ENV vars for the script to run
+
+```shell
+export IMAGE=quay.io/you-username/relations-api
+export QUAY_USER=your-quay-username
+export QUAY_TOKEN=your-quay-password
+export RH_REGISTRY_USER=your-redhat-registry-username
+export RH_REGISTRY_TOKEN=your-redhat-registry-password
+```
+
+3) Build and push the image: `make docker-build-push`
+
+**On Mac:**
+
+2) Set required ENV vars for the script to run:
+
+```shell
+export QUAY_REPO_RELATIONS=quay.io/you-username/relations-api
+```
+
+3) Login to Quay with Podman/Docker and your Quay credentials: `podman login quay.io` or `docker login quay.io`
+
+4) Build and push the image: `make build-push-minimal`
+
+
+### Building Inventory API
+1) Change to the Inventory API code path: `cd /path/to/inventory-api`
+
+**On Linux**:
+2) Set required ENV vars for the script to run
+
+```shell
+export IMAGE=quay.io/you-username/inventory-api
+export QUAY_USER=your-quay-username
+export QUAY_TOKEN=your-quay-password
+export RH_REGISTRY_USER=your-redhat-registry-username
+export RH_REGISTRY_TOKEN=your-redhat-registry-password
+```
+
+3) Build and push the image: `make docker-build-push`
+
+**On Mac:**
+2) Set required ENV vars for the script to run:
+
+```shell
+export QUAY_REPO_INVENTORY=quay.io/you-username/inventory-api
+```
+
+3) Login to Quay with Podman/Docker and your Quay credentials: `podman login quay.io` or `docker login quay.io`
+
+4) Build and push the image: `make build-push-minimal`
+
+### Building Notifications Backend
+
+1) Change to the Notifications Backend code path: `cd /path/to/notifications-backend`
+
+**All Operating Systems**
+2) Set ENV vars
+
+```shell
+export QUAY_REPO_NOTIFICATIONS=quay.io/username/notifications-backend
+export IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+```
+
+3) Login to Quay with Podman/Docker and your Quay credentials: `podman login quay.io` or `docker login quay.io`
+
+4) Build and push the image:
+
+```shell
+docker/podman build \
+  -t ${QUAY_REPO_NOTIFICATIONS}:${IMAGE_TAG} \
+  --platform linux/amd64 . \
+  -f docker/Dockerfile.notifications-backend.jvm
+
+docker/podman push ${QUAY_REPO_NOTIFICATIONS}:${IMAGE_TAG}
+```
+
+## Deploy to Ephemeral (All-in-One)
```

Comment:
Seems like VPN access is required for actually deploying to ephemeral using bonfire, a small preq somewhere should suffice. Other than that this document is extremely thorough and detailed, great job!

Reply:
Good call it, will add it!

Reply:
added VPN requirement to the initial list of prerequisites since it covers the entire doc